src/supervised/ordinal.jl

```diff
@@ -0,0 +1,49 @@
+"""
+    OrdinalMarginLoss <: SupervisedLoss
+
+Modifies a margin loss `loss` to be used on an ordinal domain with
+number of levels `nlevels`. It treats each level as an integer between
+1 and `nlevels`, inclusive, and penalizes output according to the sum of
+level thresholds crossed relative to target
+
+Assumes target is encoded in an Index encoding scheme where levels are
+numbered between 1 and `nlevels`
+"""
+struct OrdinalMarginLoss <: SupervisedLoss
+    loss::MarginLoss
+    nlevels::Int
+end
+
+for fun in (:value, :deriv, :deriv2)
+    @eval @fastmath function ($fun)(loss::OrdinalMarginLoss, target::Number, output::Number)
+        retval = 0
```

Comment:
You're absolutely right, I just checked it with `@code_warntype` and it shows that retval is of type Any. I'm thinking that I should initialize it with the type of the output, since the type instability is making the function quite slow.